### PR TITLE
Lease tests

### PR DESCRIPTION
**Pull request by jeffa5**

### CODE REVIEW

**Review by heidihoward — APPROVED** [review]

File: tests/test_single.py [hunk 1]
```diff
@@ -76,6 +76,30 @@ def test_kv_historical(http1_client):
         assert b64decode(res.json()["kvs"][0]["value"]) == f"bar{i}"
 
 
+# pylint: disable=redefined-outer-name
+def test_lease(http1_client):
+    """
+    Test that the sandbox starts.
```
Comment (jumaffre) [inline tests/test_single.py:82]: Update comment?

File: tests/test_single.py [hunk 2]
```diff
@@ -76,6 +76,30 @@ def test_kv_historical(http1_client):
...
+
+    res = http1_client.lease_keep_alive(lease_id)
+    check_response(res)
+
+    res = http1_client.lease_revoke(lease_id)
+    check_response(res)
+
+    res = http1_client.lease_keep_alive(lease_id)
+    logger.info("res: {} {}", res.status_code, res.text)
+    assert res.status_code == 400
```
Comment (jumaffre) [inline tests/test_single.py:96]: Is this because this isn't yet implemented? If so, a comment here to the GitHub issue would make it clearer.

Comment (jeffa5) [reply tests/test_single.py:100]: This is 400 because we have revoked the lease, I've added some comments.

File: src/app/app.cpp [hunk 3]
```diff
@@ -940,6 +940,11 @@ namespace app
       auto now_s = get_time_s();
       auto lstore = leasestore::LeaseStore(ctx.tx);
       auto ttl = lstore.keep_alive(id, now_s);
+      if (ttl == 0)
+      {
+        return ccf::grpc::make_error(
+          GRPC_STATUS_NOT_FOUND, "given lease has expired or has been revoked");
```
Comment (jumaffre) [inline src/app/app.cpp:946]:
```suggestion
          GRPC_STATUS_NOT_FOUND, fmt::format("given lease {} has expired or has been revoked", id));
```

**Pull request merged**